src/actions/router.js

```diff
@@ -1,14 +1,55 @@
 import parseUrl from 'grape-web/lib/parse-url'
 import * as router from 'react-router-redux'
 import find from 'lodash/collection/find'
+import {matchPath} from 'react-router-dom'
 
 import conf from '../conf'
 import * as types from '../constants/actionTypes'
 import {channelsSelector} from '../selectors'
 import {findLastUsedChannel} from './utils'
+import {channelRoute, pmRoute} from '../constants/routes'
+import {isChatUrl} from '../components/grapedown/utils'
 
 import {setChannel, openPm, openChannel, handleBadChannel} from './'
 
+const goToFromEmbedded = path => (dispatch) => {
+  const {pathname, hostname} = parseUrl(path)
+  const match = matchPath(pathname, {path: channelRoute}) ||
+    matchPath(pathname, {path: pmRoute})
+  // Open link in new tab, if it's outgoing link, or if link not leads to chat.
+  if (!match) {
+    window.open(path)
+    return
+  }
+  const {channelId, mateId, messageId} = match.params
+  const currentId = Number(channelId) || Number(mateId)
+  // Open link in the same window, because channelId of parsed link and opened chanelId similar
+  if (currentId === conf.channelId) {
```

Comment:
this is bs, mateId is never a channelId, pm channels also have a channelIds. You can not detect  from pmRoute if the url is going to be same channel or not. Just always open it here in a new window.